pandas/core/nanops.py

```diff
@@ -163,7 +166,9 @@ def _has_infs(result) -> bool:
         return False
 
 
-def _get_fill_value(dtype, fill_value=None, fill_value_typ=None):
+def _get_fill_value(
+    dtype: Dtype, fill_value: Any = None, fill_value_typ: Optional[str] = None
```

Comment:
you can be more restrictive on the fill_value, I think this can only be Optional[Scalar], we almost *always* want to avoid Any